model/annotations/task_annotations.go

```diff
@@ -313,3 +350,26 @@ func AddCreatedTicket(taskId string, execution int, ticket IssueLink, userDispla
 	)
 	return errors.Wrapf(err, "adding ticket to task '%s'", taskId)
 }
+
+// ValidateMetadataLinks will validate the given metadata links, ensuring that they are valid URLs, that they match Evergreen's approved
+// CORS origins, and that they are not too long. It also ensures that there are not more than MaxMetadataLinks links provided.
+func ValidateMetadataLinks(links ...MetadataLink) error {
+	settings := evergreen.GetEnvironment().Settings()
+	catcher := grip.NewBasicCatcher()
+	if len(links) > MaxMetadataLinks {
+		catcher.Errorf("cannot have more than %d task links per annotation", MaxMetadataLinks)
+	}
+	for _, link := range links {
+		catcher.Add(util.CheckURL(link.URL))
+		if !utility.StringMatchesAnyRegex(link.URL, settings.Ui.CORSOrigins) {
+			catcher.Errorf("task link URL '%s' must match a CORS origin", link.URL)
```

Comment:
I don't think we should do this. The CORS origins are a security feature that we use to ensure we aren't allowing different websites to directly interact with our API. We shouldn't make task annotation links be dependent on it. 
e.g. If a user wants to add a new link we shouldn't also be required to add them to our CORS origins list.

Since creating annotation links requires auth anyways that should be enough validation.

Reply:
For the parsley case which wanted to allow arbitrary files, 
Anyone can generate a link that resembles the one below. There is no auth mechanism in place for generating arbitrary links because anyone can slap together a URL and share it.
`https://parsley.mongodb.com/userSuppliedLog=someShadyLogFile.com`

For this scenario, only privileged users can set these links because they need API access in order to upload them. So the links are inherently safer because of the assumption that if a user has API access we already trust them. 

If we wanted to provide an allow list for URLs using the CORs origins is not a good idea either way. The CORs origins [are used to prevent other websites from directly making requests to Evergreen's API](https://developer.mozilla.org/en-US/docs/Glossary/CORS). So if we start expanding this list to allow any arbitrary URL that a user wants to upload as a task URL we also expose evergreen to requests from those other URLs.

Reply:
I see, I guess since this is restricted to authed users I'm fine to remove this bit. Or we could introduce an "allow" list as Mohamed mentioned but maybe that'd be best as a followup to this after we determined we need it? @ablack12

Reply:
Went ahead and removed it.